### ISSUE TITLE
Problem with random_matrix in conftest

### ISSUE THREAD

**Issue by jonbarker68** [claritychallenge/clarity#253]
The random_matrix fixture is a nice idea but it has an issue:  the seed is set in conftest (which gets called once per test module) which means that the results of the tests depend on the order that the tests are executed. This means that tests which depend on the matrix's value will fail if you try to run them individually e.g.

pytest tests/data/test_HOA_tools_cec2.py::test_P

Or will fail if a new test is added anywhere than at the end.

I propose an improved implementation that will allow both the current behaviour and one with a test-dependent seed

```python
@pytest.fixture
def random_matrix():
    """Generate a random matrix for use in tests."""
    # The fixture returns a function that can be called to generate a random matrix
    # >>> def my_test(random_matrix):
    # >>>     matrix = random_matrix(seed=1234)
    # or use the global seed
    # >>>     matrix = random_matrix() 
    # or
    def _random_matrix(seed: int | None = None) -> np.ndarray:
        if seed is not None:
            # Seed is supplied so use a generator with that seed...
            rng_to_use = np.random.default_rng(seed)
        else:
            # ... else use the global generator (defined in conftests's global namespace)
            rng_to_use = rng
        return np.asarray(rng_to_use.random((100, 100)))

    return _random_matrix
```

**Comment by jonbarker68** [comment]
The  random_matrix fixture is currently only being used in tests/data/test_HOA_tools_cec2.py so this issue can be fixed while dealing with issue #81

**Comment by jonbarker68** [comment]
This issue will be closed by PR #254
